enclave-runner/src/usercalls/mod.rs

```diff
@@ -333,36 +348,258 @@ impl EnclaveState {
             last_fd,
             exiting: AtomicBool::new(false),
             usercall_ext : usercall_ext,
+            work_sender: Mutex::new(work_sender)
         })
     }
 
+    fn create_worker_threads(num_of_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>,
+                             enclave: Arc<EnclaveState>) {
+        for _ in 1..num_of_threads {
+            let work_receiver = work_receiver.clone();
+            let io_queue_send = io_queue_send.clone();
+            let enclave = enclave.clone();
+
+            thread::spawn(move || {
+                loop {
+                    let work = work_receiver.recv();
+                    let work = match work {
+                        Err(crossbeam::channel::RecvError) => {
+                            break;
+                        },
+                        Ok(work) => work,
+                    };
+                    let res = match work {
+                        Work::Stopped(tcs, mode) => {
+                            IOHandlerInput::entry_async(enclave.clone(), tcs, &io_queue_send, mode)
+                        },
+                        Work::Running(state, usercall, coresult, mode) => {
+                            IOHandlerInput::coentry_async(state, usercall, coresult, &io_queue_send, mode)
+                        },
+                    };
+                    if let Err(err) = res {
+                        match err.0 {
+                            (_, _, EnclaveEntry::ExecutableNonMain, _) => {
+                                if let Some(cmd) = enclave.kind.as_command() {
+                                    let mut cmddata = cmd.data.lock().unwrap();
+                                    cmddata.running_secondary_threads -= 1;
+                                    if cmddata.running_secondary_threads == 0 {
+                                        cmd.wait_secondary_threads.notify_all();
+                                    }
+                                }
+                                else {
+                                    panic!("Could not extract Command from EnclaveState");
```

Comment:
Just use `unwrap` you already checked this is a `EnclaveEntry::ExecutableNonMain`